### ISSUE TITLE
Benzene becomes cyclohexane

### ISSUE THREAD

**Issue by AleksandrPokrovskii** [epam/ketcher#682]
**Steps to Reproduce**
1. Launch Ketcher
2. Select the Benzene template
3. Add the Benzene
4. Click to the Benzene bond
5. Click to the common bond

**Expected behavior**
The structure does not change
![image](https://user-images.githubusercontent.com/85170103/126496377-29a3bfa3-9f28-4e59-b3f7-e285fc676d25.png)

**Actual behavior**
The first benzene becomes cyclohexane
![image](https://user-images.githubusercontent.com/85170103/126496436-37b0eeee-1e14-4327-9082-6692348bc7f3.png)

**Comment by AleksandrPokrovskii** [comment]
It turned out that the addition of benzene to benzene is incorrect. @AlexanderSavelyev Please write the rules for adding benzene

**Comment by karen-sarkisyan** [comment]
An update on issue after analysis.
**TL;DR**: further enhancements and refactoring are needed, so this issue is not likely to be fixed in 2.4 release.

**Explanation**
In order to fuse aromatic rings we used aromatize/dearomatize functions up until version 2.3

An issue was introduced after we stopped reversing order of operations with [this commit](https://github.com/epam/ketcher/commit/cbdf1e0). It prevented correct working of Undo/Redo actions for merged aromatic rings.

Currently we avoid this issue by avoiding aromatize/dearomatize cycle altogether, fusing aromatic rings as regular structures:
https://github.com/epam/ketcher/blob/3b9183363aadbecc15dd5f7bcc848401342fe0ca/packages/ketcher-core/src/application/editor/actions/aromaticFusing.ts#L50-L53
, which in turn introduces the issue of Benzene accidentally turning into Cyclohexane on 2nd click on bond.

So, the following work needs to be done:
- Stabilization of data exchange with Indigo endpoint in `ket` format, including molecule autoloader for aromatize/dearomatize: [Issue 526 in Indigo](https://github.com/epam/Indigo/issues/562)
- Refactoring of `fromAromaticTemplateOnBond` action creator to fix issues with operations order in history stack
- Handling edge case of structures that lose aromaticity upon fusing, like adding 3rd benzene ring to naphthalene:
 
![image](https://user-images.githubusercontent.com/18420141/147102427-e4bea15d-ea4f-454b-92b4-7c4097a1a30a.png)
